src/EditorFeatures/Core/Implementation/InlineRename/InlineRenameSession.OpenTextBufferManager.cs

```diff
@@ -459,6 +413,49 @@ internal void ApplyConflictResolutionEdits(IInlineRenameReplacementInfo conflict
                 }
             }
 
+            private static async Task<IEnumerable<TextChange>> GetTextChangesFromTextDifferencingServiceAsync(Document oldDocument, Document newDocument, CancellationToken cancellationToken = default(CancellationToken))
+            {
+                try
+                {
+                    using (Logger.LogBlock(FunctionId.Workspace_Document_GetTextChanges, newDocument.Name, cancellationToken))
+                    {
+                        if (oldDocument == newDocument)
+                        {
+                            // no changes
+                            return SpecializedCollections.EmptyEnumerable<TextChange>();
+                        }
+
+                        if (newDocument.Id != oldDocument.Id)
+                        {
+                            throw new ArgumentException(WorkspacesResources.DocumentVersionIsDifferent);
+                        }
+
+                        SourceText oldText = await oldDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+                        SourceText newText = await newDocument.GetTextAsync(cancellationToken).ConfigureAwait(false);
+
+                        if (oldText == newText)
+                        {
+                            return SpecializedCollections.EmptyEnumerable<TextChange>();
+                        }
+
+                        var textChanges = newText.GetTextChanges(oldText).ToList();
```

Comment:
SourceText.GetChanges is allowed to return the entire source as a single change, as opposed to actually attempting a text diff. The intent is not to produce a diff, but to access the known changes between two source texts, assuming they are a linear progression from one to another.   

Document.GetTextChanges attempts to use SourceText.GetChanges to get at known deltas in the underlying buffer.  If that produces nothing or just a single change, then Document.GetTextChanges does fall back to attempting to diff the documents.  However, it uses the tree diffing algorithm and not a text diffing algorithm, because its generally faster, and being a different algorithm may produce different results.